### DISCUSSION TITLE
classes/class_filedialog

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#206] [User-contributed notes]
# classes/class_filedialog

Inherits: ConfirmationDialog< AcceptDialog< Window< Viewport< Node< Object A dialog for selecting files or directories in the filesystem. Description: FileDialog is a preset dialog used to choose f...

https://docs.godotengine.org/en/latest/classes/class_filedialog.html

**Comment by llama-nl** [comment]
Does the `use_native_dialog` work for Android and iOS? Also will it work on Android Editor and iOS.

**Reply by Calinou** [reply]
Native file dialogs are currently not implemented on Android and iOS, as evidenced by https://github.com/godotengine/godot-proposals/issues/10983.

**Comment by acarlstein** [comment]
Greetings, thank you for managing this open source project. Much appreciated.

Are there examples showing how this is used and how to use its properties?

**Reply by Calinou** [reply]
Like other AcceptDialog-based nodes, you typically use FileDialog by adding it as a node in the scene, then calling `popup_centered_ratio()` on the FileDialog from a script when you want it to show up. Connect the relevant FileDialog signal (such as `file_selected`) to do something with the file name that was chosen by the user (such as reading the file in question).

**Comment by PublicFacingAccount** [comment]
It seems like FileDialog cares about what its parent is. I was getting a blank/empty/unusable window when I called file_dialog.popup(), when FileDialog was a child of another Popup (in my case, a PopupMenu). But it displayed as expected when it was a child of a Control. I don't know if this is a property of all Windows? Or if this is written down somewhere? I couldn't find any information about it.
Leaving a comment in case it helps someone.

**Comment by morningkingdom** [comment]
hi there, is there a way to see show the thumbnails while using FileDialog?

**Reply by Calinou** [reply]
Thumbnail previews aren't supported in FileDialog, but they are in EditorFileDialog (which can only be used in the editor).

See this proposal which would allow using EditorFileDialog functionality in exported projects:

- https://github.com/godotengine/godot-proposals/issues/6831

**Comment by ondesic** [comment]
Is there a way to make the Native Dialog be exclusive, like the Godot FileDialog is? I actually prefer the native dialog and am grateful it is an option, but the user can click outside of the dialog and make it close without a file selected, EVEN if exclusive is selected.
Here is the problem, the `await` function is still waiting and is in limbo. Do I need to create a bug report?

**Reply by Calinou** [reply]
This is a bug, as described in this issue:

- https://github.com/godotengine/godot/issues/85047

**Reply by ondesic** [reply]
Calinou, the posters on that tread feel that this is a different bug. I guess I'll open a separate bug report?

**Comment by SamDevelopsCode** [comment]
If you want a simple file selection window popup (and not the save file window that is by some weird decision the default) use the following line of code -

```dialog.file_mode = FileDialog.FILE_MODE_OPEN_FILE```

If I was a better dev, I'd make a Documentation PR. Sadly I am a lesser dev, so I shall not.
